src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/CSharpConvertToRecordRefactoringProvider.cs

```diff
@@ -217,24 +220,33 @@ private static SyntaxList<AttributeListSyntax> GetModifiedAttributeListsForPrope
         /// modifies constructors and some method modifiers to fall in line with record requirements (e.g. this() initializer)
         /// </summary>
         /// <param name="typeDeclaration">Original type declaration</param>
+        /// <param name="type"></param>
         /// <param name="semanticModel">Semantic model</param>
-        /// <param name="propertiesToMove">Properties we decided to move</param>
-        /// <param name="cancellationToken">Cancellation token</param>
-        /// 
+        /// <param name="propertiesToMove">Properties we decided to move, may return in a different order</param>
+        /// <param name="defaults">properties in order for positional record with associated defaults</param>
+        /// <param name="cancellationToken">cancellationToken</param>
         /// <returns>The list of members from the original type, modified and trimmed for a positional record type usage</returns>
         private static ImmutableArray<MemberDeclarationSyntax> GetModifiedMembersForPositionalRecord(
             TypeDeclarationSyntax typeDeclaration,
             INamedTypeSymbol type,
             SemanticModel semanticModel,
             ImmutableArray<PositionalParameterInfo> propertiesToMove,
+            out ImmutableArray<(PropertyDeclarationSyntax property, EqualsValueClauseSyntax? @default)> defaults,
```

Comment:
Is this an array versus a dictionary intentionally, as could we see the same property twice?

Reply:
The main thing I want to do here is maintain the order. I'll clean it up by maintaining the property list order as the same as the default list, then I can zip the two together when I need both.